### PR TITLE
py3: fixes test_DatasetManager.py:DatasetManagerTestCase

### PR DESCRIPTION
**Pull request by caleb-easterly**
Looks like there was a plan to cast/extract to list anyway?

xref. #1715

### CODE REVIEW

File: lib/galaxy/managers/rbac_secured.py [hunk 1]
```diff
@@ -87,16 +87,15 @@ def __assert_action(self):
...
-    # TODO: list?
     def by_roles(self, dataset, roles):
         permissions = self.by_dataset(dataset)
-        return filter(lambda p: p.role in roles, permissions)
+        return list(filter(lambda p: p.role in roles, permissions))
 
     def by_role(self, dataset, role):
         permissions = self.by_dataset(dataset)
-        found = filter(lambda p: p.role == role, permissions)
+        found = list(filter(lambda p: p.role == role,  permissions))
```
Comment (mvdbeek) [inline lib/galaxy/managers/rbac_secured.py]: The extra space will make the linting fail

**Review by mvdbeek — APPROVED** [review]

File: lib/galaxy/managers/rbac_secured.py (hunk 1 above)
Comment (caleb-easterly) [reply lib/galaxy/managers/rbac_secured.py]: good eye, thanks!

**Comment by mvdbeek** [comment]
I had a similar change in https://github.com/galaxyproject/galaxy/pull/6450 and the tests passed there, so I'll merge this now.

**Comment by nsoranzo** [comment]
For future reference, list comprehensions would make the code clearer IMO.

**Pull request merged**